### PR TITLE
Scheduler talks to TPs directly with SSL enabled in dev environment

### PR DESCRIPTION
**Pull request by Sindica**
. Pick up yunwen's scheduler changes.
. Generate RP kubeconfig for scheduler in TP - with SSL enabled
. Works in 2TP/1RP

### CODE REVIEW

**Comment by yb01** [comment]
please rebase with master to get the scheduler code changes

File: docs/setup-guide/scale-out-local-dev-setup.md [hunk 1]
```diff
@@ -34,23 +34,40 @@ export RESOURCE_PARTITION_IP=[RP_IP]
...
 
 1. Expected last line of output: "Tenant Partition Cluster is Running ..."
 
+Note:
+
+1. As we start to picking up secure mode in scale out, api server certificates will be shared across all partitions in 
+development environment. The first TP that started needs to generate api server certificates and be copied over to other
+TP/RP before they start.
+
+1. To generate first set of certs, run `REUSE_CERTS=false; ./hack/arktos-up-scale-out-poc.sh`
```
Comment (yb01) [inline docs/setup-guide/scale-out-local-dev-setup.md:54]: the IPs ( or master name ) for the other cluster's master will needed to set the proper SANs or the certs.

File: hack/lib/util.sh [hunk 2]
```diff
@@ -507,22 +507,15 @@ function kube::util::write_client_kubeconfig {
...
+    local protocol=${8:-"https"}
 
     cat <<EOF | ${sudo} tee "${dest_dir}"/"${client_id}".kubeconfig > /dev/null
 apiVersion: v1
 kind: Config
 clusters:
   - cluster:
       certificate-authority: ${ca_file}
-      server: ${MASTER_ENDPOINT}
+      server: ${protocol}://${api_host}:${api_port}/
```
Comment (yb01) [inline hack/lib/util.sh:518]: do we need the tailing "/"

Comment (Sindica) [reply hack/lib/util.sh:518]: This is the original code

Comment (Sindica) [reply hack/lib/util.sh:518]: I added protocol

File: docs/setup-guide/scale-out-local-dev-setup.md (hunk 1 above)
Comment (Sindica) [reply docs/setup-guide/scale-out-local-dev-setup.md:54]: This is user manual. It's better let the tool handle the terminology. I would avoid add technical terms here.

**Pull request merged**